### PR TITLE
setup: Make sure Raspberry Pi OS 'pi' user has default 'raspberry' password

### LINKED ISSUE #330: 'pi' user handling on latest Raspberry Pi OS images
The latest (from _2022-04-04_ on) Raspberry Pi OS images don't have anymore default password `raspberry` for default user `pi`: user `pi` (which still exists) has **no** password by default.
As a result, in a headless setup, it is no more possible to login as `pi`, unless user/password has been configured in some way. See:
- https://downloads.raspberrypi.org/raspios_lite_armhf/release_notes.txt  (_2022-04-04_ entry)
- https://www.raspberrypi.com/news/raspberry-pi-bullseye-update-april-2022/
- https://www.raspberrypi.com/documentation/computers/configuration.html#configuring-a-user

For Pynab images, this will have to be handled in the Pynab `setup` script.
The easiest way (to emulate current setup of Pynab images) is to generate a `/boot/userconf.txt` file containing the line:`pi:<password>`, `<password>` being the encrypted form of `raspberry` password.

### PR DESCRIPTION
**Pull request by f-laurens**
Resolves #330.

### CODE REVIEW

**Comment by f-laurens** [comment]
**Note**: CI is currently broken for raspios "latest" images, causing checks with these images to fail...
See issue https://github.com/nabaztag2018/pynab/issues/329.

File: setup [hunk 1]
```diff
@@ -111,24 +111,30 @@ then
...
         # be kind to users:
+        if [ ! -f "/boot/userconf.txt" ]
+        then
+            # make sure 'pi' user has default 'raspberry' password
+            echo 'pi:$y$j9T$9ZU7FXeU7PVHm49rB5Hbe.$MHZMLJfNfXBFbYgGwg/TNjADIRbna.Epxzd/MrqPJM9' > '/boot/userconf.txt'
+        fi
         if [ ! -f "/boot/ssh" ]
         then
+            # enable SSH
             touch /boot/ssh
```
Comment (pguyot) [inline setup:133]: Truth is we shouldn't enable SSH by default.

File: setup [hunk 2]
```diff
@@ -111,24 +111,30 @@ then
...
-        # assume Rasberry Pi OS distro
+    elif [ -f "/etc/rpi-issue" ]
+    then
+        # Raspberry Pi OS distro
         hostname="raspberrypi"
         # be kind to users:
+        if [ ! -f "/boot/userconf.txt" ]
+        then
+            # make sure 'pi' user has default 'raspberry' password
+            echo 'pi:$y$j9T$9ZU7FXeU7PVHm49rB5Hbe.$MHZMLJfNfXBFbYgGwg/TNjADIRbna.Epxzd/MrqPJM9' > '/boot/userconf.txt'
```
Comment (pguyot) [inline setup:128]: We shouldn't put this default password either.

File: setup (hunk 1 above)
Comment (f-laurens) [reply setup:133]: > enable SSH by default

This is done since #283 was merged.

File: setup (hunk 2 above)
Comment (f-laurens) [reply setup:128]: Two options then:

- use another secret default password, known only to Nabaztag owners;
- let users cope by themselves with setting up the password.
We may point them to  the [Raspberry PI imager](https://www.raspberrypi.com/software/), which now allows  configuring (at least for Raspberry Pi OS based images) image settings: Wifi setup, SSH setup, user/password.

**Comment by f-laurens** [comment]
Assume users will handle this by themselves. Closing.

**Pull request closed without merging**